code/modules/projectiles/guns/projectile/constructable/swords.dm

```diff
@@ -79,15 +79,33 @@
 	icon_state = "venom_sword"
 	var/beaker = null
 	var/obj/item/weapon/reagent_containers/hypospray/HY = null
-	var/max_beaker_volume = 50 //The maximum volume a beaker can have and still be placed into the sword
-	var/inject_amount = 5 //The amount of reagents injected from the beaker each hit
+	var/max_beaker_volume = 500 //The maximum volume a beaker can have and still be placed into the sword
+	var/min_inject_amount = 5
+	var/max_inject_amount = 20
+	var/inject_amount = 20 //The amount of reagents injected from the beaker each hit
+
+/obj/item/weapon/sword/venom/verb/set_inject_amount()
+	set name = "Set injection amount"
+	set category = "Object"
+	set src in range(0)
+	if(usr.incapacitated())
+		return
+	var/N = input("Injection amount:","[src]") as null|num
```

Comment:
You always need sanity after an input - is the user still in range, are they still not incapacitated, etc